RealmSwift/Sync.swift

```diff
@@ -895,8 +1068,18 @@ extension User {
 
      @return A `Realm.Configuration` instance with a flexible sync configuration.
      */
-    public func flexibleSyncConfiguration() -> Realm.Configuration {
-        let config = self.__flexibleSyncConfiguration()
+    public func flexibleSyncConfiguration(clientResetMode: ClientResetMode = .recoverUnsyncedChanges(nil, nil)) -> Realm.Configuration {
+        var config: RLMRealmConfiguration
+        switch clientResetMode {
```

Comment:
Can we extract this to a helper function given that this is done here, the flx initialiser with initial subscription  and in the PBS initialiser

Reply:
I think we need to use the mostly repetitive callsites because we need to extract the client reset blocks.

If we move them to the same method I don't think we'd save much repetition. Three different objc methods are called (w/ partition, w/ flex, w/ flex and initial subs). There'd still be 15 cases, each method with 5 modes. The only difference is that all 15 cases would be in one method.